src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -3335,6 +3703,26 @@ opendir(const char *path)
 		D_GOTO(out_err, rc = ENAMETOOLONG);
 	}
 
+	if (compatible_mode) {
+		/* add fd kernel to hash table */
+		D_ALLOC_PTR(fd_ht_obj);
+		if (fd_ht_obj == NULL) {
+			/* not returning ENOMEM since dirp_kernel is valid */
+			free_dirfd(idx_dirfd);
+			goto out_compatible;
+		}
+		fd_ht_obj->real_fd = dirfd(dirp_kernel);
+		assert(fd_ht_obj->real_fd >= 0);
+		fd_ht_obj->fake_fd = idx_dirfd + FD_DIR_BASE;
+		rc = d_hash_rec_insert(fd_hash, &fd_ht_obj->real_fd, sizeof(int), &fd_ht_obj->entry,
+				       true);
```

Comment:
Again, you set `exclusive` here which has a performance impact, is that required?

Reply:
Fixed as you suggested. Thank you!